src/components/AccountPage/AccountHeader.js

```diff
@@ -29,26 +77,56 @@ const AccountName = styled(Text).attrs({
 
 type Props = {
   account: TokenAccount | Account,
+  parentAccount: ?Account,
+  t: T,
 }
 
-class AccountHeader extends PureComponent<Props> {
-  render() {
-    const { account } = this.props
+const AccountHeader: React$ComponentType<Props> = React.memo(
+  ({ account, parentAccount, t }: Props) => {
     const currency = getAccountCurrency(account)
+    const mainAccount = getMainAccount(account, parentAccount)
+    const explorerView = getDefaultExplorerView(mainAccount.currency)
+
+    const getContract = () =>
+      account.type === 'TokenAccount' && parentAccount
+        ? getAccountContractExplorer(explorerView, account, parentAccount)
+        : null
+
+    const contract = getContract()
+
+    const openLink = () => {
+      contract ? openURL(contract) : null
+    }
+
     return (
       <Box horizontal align="center" flow={2} grow>
         <Box>
           <ParentCryptoCurrencyIcon currency={currency} borderColor="lightGrey" />
         </Box>
         <Box grow>
-          <CurName>{account.type === 'Account' ? currency.name : 'token'}</CurName>
+          {contract && account.type === 'TokenAccount' ? (
+            <Box horizontal alignItems="center">
+              <CurNameToken>{`${t('account.contractAddress')}`}</CurNameToken>
```

Comment:
you could use `<Trans` to not have to pass in `t={t}`

Reply:
done